provider/dummy/environs.go

```diff
@@ -233,6 +236,7 @@ type environProvider struct {
 	apiPort                    int
 	controllerState            *environState
 	state                      map[string]*environState
+	db                         coredatabase.TrackedDB
```

Comment:
I really hate having to mess about with the dummy environ. It should be strangled out.

Reply:
100% yes. You're not actually testing what you think you're testing at this level.